test/test_torch.py

```diff
@@ -4427,6 +4427,122 @@ def test_diff(self, device, dtype):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.long, torch.float32, torch.complex64)
+    def test_gradient_tensor_list(self, device, dtype):
+        inputs = (
+            ((4, 4, 4), [(1., 3.0, 6., 10.), (1., 3.0, 4., 90.), (1., 2.5, 3.4, 7.)], [0, 1, 2], 1),
+            ((4, 4, 4, 3), [(1., 3.0, 6., 10.), (1., 2.5, 7.)], [0, -1], 1),
+            ((4, 5, 2, 4, 3), [(1., 3.0, 6., 10.), (1., 3.4, 7.)], [-2, -1], 1),
+            ((4, 3, 6, 5, 3), [(1., 3.0, 6., 10.), (2.5, 3.4, 7.)], [0, 1], 1),
+            ((4, 3, 2, 5, 3), [(1., 3.0), (3.4, 7., 6.0)], [2, 1], 1),
+        )
+
+        for size, coordinates, dim, edge_order in inputs:
+            for contig in [True, False]:
+                t = make_tensor(size, device, dtype, noncontiguous=not contig)
+                t_n = t.cpu().numpy()
+                coordinates_list = []
+                for space in coordinates:
+                    if dtype == torch.complex64 and isinstance(space, float) and abs(space) < 1e-7:
+                        return
+                    coordinates_list.append(torch.tensor(space, dtype=float, device=device))
+                result = torch.gradient(t, domain=coordinates_list, dim=dim, edge_order=edge_order)
+                result_n = np.gradient(t_n, *coordinates, axis=dim, edge_order=edge_order)
+                if not isinstance(result_n, list):
```

Comment:
add a comment here explaining the wrapping

Reply:
Optionally: create a "reference_gradient" helper function with the same signature as torch.gradient that calls np.gradient and performs this conversion